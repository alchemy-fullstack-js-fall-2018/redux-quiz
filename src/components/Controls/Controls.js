import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Controls = ({
  nameValue,
  onNameInput,
  onGreetingInput,
  greetingValue,
  nameLabel,
  greetingLabel
}) => {
  return (
    <Fragment>
      <label>&nbsp;{greetingLabel}&nbsp;</label>
      <input value={greetingValue} onChange={onGreetingInput} />
      <label>&nbsp;{nameLabel}&nbsp;</label>
      <input value={nameValue} onChange={onNameInput} />
    </Fragment>
  );
};

Controls.propTypes = {
  nameValue: PropTypes.string.isRequired,
  onNameInput: PropTypes.func.isRequired,
  nameLabel: PropTypes.string.isRequired,
  greetingLabel: PropTypes.string.isRequired,
  greetingValue: PropTypes.string.isRequired,
  onGreetingInput: PropTypes.func.isRequired
};

export default Controls;
