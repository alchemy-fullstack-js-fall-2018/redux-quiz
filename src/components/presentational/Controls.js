import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ name, greeting, changeName, changeGreeting }) => {


  return (
    <section>
      <label htmlFor="name">Name</label>
      <input
        name="name" type="text"
        value={name} onChange={changeName}
      ></input>
      <label htmlFor="greeting">Greeting</label>
      <input
        name="greeting" type="text"
        value={greeting} onChange={changeGreeting}
      ></input>
    </section>
  );
};

Controls.propTypes = {
  name: PropTypes.string.isRequired,
  greeting: PropTypes.string.isRequired,
  changeName: PropTypes.func.isRequired,
  changeGreeting: PropTypes.func.isRequired
};

export default Controls;
