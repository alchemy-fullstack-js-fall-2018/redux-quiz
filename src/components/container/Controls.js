import { connect } from 'react-redux';
import Controls from '../presentational/Controls';
import { changeName } from '../../state/actions/name';
import { changeGreeting } from '../../state/actions/greeting';

const mapDispatchToProps = dispatch => ({
  changeGreeting: greeting => dispatch(changeGreeting(greeting)),
  changeName: name => dispatch(changeName(name))
});

export default connect(
  null, mapDispatchToProps
)(Controls);
