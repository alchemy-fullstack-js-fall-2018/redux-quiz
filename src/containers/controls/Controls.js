import { connect } from 'react-redux';
import Controls from '../../components/controls/Controls';
import { getGreeting } from '../../selectors/greeting';
import { greetingChange } from '../../actions/greeting';

const mapStateToProps = state => ({
  text: getGreeting(state)
});

const mapDispatchToProps = dispatch => ({
  onTextChange: ({ target }) => dispatch(greetingChange(target.value))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Controls);
