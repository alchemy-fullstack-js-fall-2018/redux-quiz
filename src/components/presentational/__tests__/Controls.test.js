import React from 'react';
import { shallow } from 'enzyme';
import Controls from '../Controls';

describe('<Controls />', () => {
  test('renders', () => {
    const changeName = jest.fn();
    const changeGreeting = jest.fn();
    const name = 'the dude';
    const greeting = 'abide, sir';
    const wrapper = shallow(
      <Controls
        name={name}
        greeting={greeting}
        changeName={changeName}
        changeGreeting={changeGreeting}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
