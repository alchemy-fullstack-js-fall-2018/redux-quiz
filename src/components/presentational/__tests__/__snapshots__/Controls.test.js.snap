// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> renders 1`] = `
<section>
  <label
    htmlFor="name"
  >
    Name
  </label>
  <input
    name="name"
    onChange={[MockFunction]}
    type="text"
    value="the dude"
  />
  <label
    htmlFor="greeting"
  >
    Greeting
  </label>
  <input
    name="greeting"
    onChange={[MockFunction]}
    type="text"
    value="abide, sir"
  />
</section>
`;
